igvc_navigation/src/mapper/map_utils.cpp

```diff
@@ -168,6 +169,54 @@ void getEmptyPoints(const pcl::PointCloud<pcl::PointXYZ>& pc, pcl::PointCloud<pc
   }
 }
 
+void removeOccupiedFromImage(cv::Mat& image, const PointCloud& last_scan,
+                             const image_geometry::PinholeCameraModel& camera_model,
+                             const tf::Transform& camera_to_odom, const RemoveOccupiedOptions& options)
+{
+  tf::Transform odom_to_camera = camera_to_odom.inverse();
+  cv::Rect image_rect = cv::Rect{ 0, 0, image.size().width, image.size().height };
+
+  cv::Mat projected_image(image.size().height, image.size().width, image.type(), static_cast<uchar>(0));
+
+  std::vector<cv::Point2d> projected_points;
+  projected_points.reserve(100);
+
+  // Transform pointcloud to image frame, add if inside image
+  for (const auto& point : last_scan.points)
+  {
+    tf::Vector3 transformed = odom_to_camera * tf::Vector3{ point.x, point.y, 0 };
```

Comment:
Is Z already removed or do we have access to it here? I would just explain why it is ignored in a comment.